index.html

```diff
@@ -156,9 +158,80 @@ <h2><dfn>SmartCardContext</dfn> interface</h2>
+          <td>Whether there is an ongoing PC/SC operation in this context.</td>
+        </tr>
+      </table>
       <section>
         <h3><dfn>listReaders()</dfn> method</h3>
-        <div class="issue">Write an algorithm for this method.</div>
+        <p>The {{SmartCardContext/listReaders()}} method steps
+        are:</p>
+        <ol>
+          <li>If [=this=].{{SmartCardContext/[[operationInProgress]]}} is
+            `true`, [=exception/throw=] a "{{InvalidStateError}}"
+            {{DOMException}}.</li>
+          <li>Set [=this=].{{SmartCardContext/[[operationInProgress]]}} to
+            `true`.</li>
+          <li>Let |promise:Promise| be [=a new promise=].</li>
+          <li>Run the following steps [=in parallel=]:
+            <ol>
+              <li>Let |resourceQuery:RESOURCEQUERY| be a new instance of the
+                platform's [[PCSC5]] `RESOURCEQUERY` class, with
+                [=this=].{{SmartCardContext/[[resourceManager]]}} as its
+                contructor input parameter.</li>
+              <! –– NB: Can't have "[]" in the data type part of a variable declaration. ––>
+              <li>Let |groups:array of STR| be the platform's [[PCSC5]] `STR[]`
+                containing the list of groups names that is equivalent to "all
+                readers in the system" in that platform.
+                <aside class="note">This could also be an empty `STR[]` in some
+                  [[PCSC5]] implementations.</aside>
+              </li>
+              <li>Let |pcscReaders:array of STR| be an empty `STR[]`.
+              <li>Invoke the `ListReaders` method of |resourceQuery| with
+                |groups| as input and |pcscReaders| as output parameters.</li>
+              <li>Let |responseCode:RESPONSECODE| be the returned
+                `RESPONSECODE`.</li>
+              <li>Destroy |resourceQuery|.</li>
+              <li>If |responseCode| is not `SCARD_S_SUCCESS`, [=queue a global
+                task=] on the [=relevant global object=] of [=this=] using the
+                [=smart card task source=] which performs the following
+                steps:
+                <ol>
+                  <li>Set [=this=].{{SmartCardContext/[[operationInProgress]]}} to
+                    `false`.</li>
+                  <li>[=Reject=] |promise| with a {{DOMException}}
+                    {{SmartCardError/corresponding}} to |responseCode|.</li>
+                </ol>
+              </li>
+              <li>Otherwise, [=queue a global task=] on the [=relevant global
+                object=] of [=this=] using the [=smart card task source=] which
+                performs the following steps:
+                <ol>
+                  <li>Set [=this=].{{SmartCardContext/[[operationInProgress]]}} to
+                    `false`.</li>
+                  <li>[=resolve=] |promise| with a `sequence` of {{DOMString}}
+                    equivalent to |pcscReaders|.</li>
+                </ol>
+              </li>
```

Comment:
Refactor this since you always queue a global tasks and set `operationInProgress` to `false`.

Reply:
Done.